lib/hooks/profile.ts

```diff
@@ -0,0 +1,34 @@
+import useSWR, { mutate } from "swr";
+import IUser from "../../types/IUser";
+
+const fetcher = (url: string) => fetch(url).then((r) => r.json());
+
+/**
+ * Gets the profile of another user by their id.
+ */
+export const useProfile = (id: string) => {
+  const { data, error } = useSWR<IUser>(`/api/user/${id}`, fetcher);
```

Comment:
This looks good! I think it was a good idea to get users by their unique ID instead of their email address or something else.